sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala

```diff
@@ -663,14 +666,13 @@ class ResolveSessionCatalog(
   object TempViewOrV1Table {
     def unapply(nameParts: Seq[String]): Option[Seq[String]] = nameParts match {
       case _ if isTempView(nameParts) => Some(nameParts)
-      case SessionCatalogAndTable(_, tbl) =>
-        if (nameParts.head == CatalogManager.SESSION_CATALOG_NAME && tbl.length == 1) {
-          // For name parts like `spark_catalog.t`, we need to fill in the default database so
-          // that the caller side won't treat it as a temp view.
-          Some(Seq(catalogManager.v1SessionCatalog.getCurrentDatabase, tbl.head))
-        } else {
-          Some(tbl)
+      case Seq(CatalogManager.SESSION_CATALOG_NAME, name @ _*) if isTempView(name) =>
+        throw new AnalysisException(s"Table or view not found: ${nameParts.quoted}")
```

Comment:
I am failing here early for temp view as well instead of returning `Some(nameParts)` so that the resolution rule is consistent. For example, for commands like `REFRESH TABLE`, they exit with `Database 'session_catalog' not found;` if we don't fail early here.

Reply:
For Spark 2.4, yes we will fail with `Database 'spark_catalog' not found`. But in 3.0, we have a new resolution rule, and it's more reasonable to fail with something like `invalid namespace for session catalog` when we see `spark_catalog.t`.

Reply:
If we fail early here, we will not see `Database 'session_catalog' not found;` which happens when the command for `REFRESH TABLE` runs.

Do you mean to update the exception message here?

Reply:
Yea, we should update the error message to report invalid namespace rather than "database not found", when seeing `spark_catalog.t`.